thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/anomaly/detection/trigger/DataAvailabilityTaskScheduler.java

```diff
@@ -176,35 +198,43 @@ private void populateDetectionMapAndDatasetConfigMap(
     return res;
   }
 
-  private long createDetectionTask(DetectionConfigDTO detectionConfig) throws JsonProcessingException {
-    long detectionConfigId = detectionConfig.getId();
+  private DetectionPipelineTaskInfo getDetectionPipelineTaskInfo(DetectionConfigDTO detectionConfig, long end) {
     // Make sure start time is not out of DETECTION_TASK_MAX_LOOKBACK_WINDOW
-    long end = System.currentTimeMillis();
     long start = Math.max(detectionConfig.getLastTimestamp(),
         end - ThirdEyeUtils.DETECTION_TASK_MAX_LOOKBACK_WINDOW);
-    DetectionPipelineTaskInfo taskInfo = new DetectionPipelineTaskInfo(detectionConfigId, start, end);
+
+    return new DetectionPipelineTaskInfo(detectionConfig.getId(), start, end);
+  }
+
+  private long createTask(TaskConstants.TaskType taskType, DetectionConfigDTO detectionConfig, long end)
+      throws JsonProcessingException {
+    DetectionPipelineTaskInfo taskInfo = getDetectionPipelineTaskInfo(detectionConfig, end);
     String taskInfoJson = OBJECT_MAPPER.writeValueAsString(taskInfo);
-    TaskDTO taskDTO = new TaskDTO();
-    taskDTO.setTaskType(TaskConstants.TaskType.DETECTION);
-    taskDTO.setJobName(TaskConstants.TaskType.DETECTION.toString() + "_" + detectionConfigId);
-    taskDTO.setStatus(TaskConstants.TaskStatus.WAITING);
-    taskDTO.setTaskInfo(taskInfoJson);
-    long taskId = taskDAO.save(taskDTO);
-    LOG.info("Created detection pipeline task {} with taskId {}", taskDTO, taskId);
-    taskLastCreateTimeMap.put(detectionConfigId, end);
-    return taskId;
+    TaskDTO taskDTO = TaskUtils.buildTask(detectionConfig.getId(), taskInfoJson, taskType);
+    long id = taskDAO.save(taskDTO);
+    LOG.info("Created {} task {} with taskId {}", taskType, taskDTO, id);
+    return id;
+  }
+
+  private long createDetectionTask(DetectionConfigDTO detectionConfig, long end) throws JsonProcessingException {
+    return createTask(TaskConstants.TaskType.DETECTION, detectionConfig, end);
+  }
+
+  private long createDataAvailabilityTask(DetectionConfigDTO detectionConfig, long end) throws JsonProcessingException {
```

Comment:
"DataAvailability" is a heavily overloaded term now.
We have DataAvailabilityTaskScheduler which is actually "EventTriggerTaskScheduler".
Here we have another "data availability" task. It is too confusing.
Can you do a refactoring?
Also @vincentchenjl  FYI

Reply:
@vincentchenjl agreed. Let's use data availability term for referring to the event (data available or not) and use Data SLA for the alerts that are setup based on these data availability events.

I will update accordingly.

Reply:
I like the term SLA since it is easy to understand.